packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1981,6 +1940,10 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
       extentOffset: extentOffset,
       affinity: fromPosition.affinity,
     );
+    final bool focusingEmpty = newSelection.baseOffset == 0 && newSelection.extentOffset == 0 && !hasFocus;
+    if (newSelection == selection && cause != SelectionChangedCause.keyboard && !focusingEmpty) {
```

Comment:
why does it need to check for other condition if the selection is the same?

Reply:
I am not sure, I found this change that introduced this: https://github.com/flutter/flutter/pull/29563
cc @mdebbar 
It was only calling onSelectionChanged under those circumstances.

Without this change, this test from that change fails:
https://github.com/flutter/flutter/blob/9cfaabaed09e1c6606c9148b0f716a78146ec4e6/packages/flutter/test/rendering/editable_test.dart#L725


However, I notice it now causes another failure, so there may be a bug here. One test was expecting one thing from the old API, and the other test is expecting something else from the new API.

Reply:
I assumed the logic in `_handleSelectionChange` was necessary, so I carried it over to the new API. It probably was modified by other changes after #29563. Let me see if removing the keyboard and focus part still passes the existing test.

Reply:
The `!focusingEmpty` was added in https://github.com/flutter/flutter/pull/49000.

The `cause != SelectionChangedCause.keyboard` was added in https://github.com/flutter/flutter/pull/42879.

Reply:
🙏  Thanks for tracing those back @mdebbar! So I think this is indeed intentional behavior, I am looking at the other test now that is broken here by carrying the logic over.

Reply:
https://github.com/flutter/flutter/pull/29809/files is where the test came from, the cupertino one is broken, while the same material test passes, so we may be missing something in cupertino text field.